x-pack/plugins/security_solution/common/endpoint/models/ecs_safety_helpers.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ECSField } from '../types';
+
+/**
+ * Use these functions to accecss information held in `ECSField`s.
+ */
+
+/**
+ * True if the field contains `expected`. If the field contains an array, this will be true if the array contains `expected`.
+ */
+export function hasValue<T>(valueOrCollection: ECSField<T>, expected: T): boolean {
+  if (Array.isArray(valueOrCollection)) {
+    return valueOrCollection.includes(expected);
+  } else {
+    return valueOrCollection === expected;
+  }
+}
+
+/**
+ * Return first non-null value. If the field contains an array, this will return the first value that isn't null. If the field isn't an array it'll be returned unless it's null.
+ */
+export function firstValue<T>(valueOrCollection: ECSField<T>): T | undefined {
```

Comment:
:grey_question: As a consumer, I would be surprised by this if there were a null in the 1st array position and I didn't get that back from a function called `firstValue`. Suggested: `oneValue`, `firstNonNullValue`, ``

Reply:
lets do `firstNonNullValue` for now and we can change it later if thats annoying to read